### DISCUSSION TITLE
一分钟读论文：《Montage：基于神经网络语言模型 (NNLM) 实现 JS 引擎模糊测试器》

### DISCUSSION THREAD

**Discussion by giscus** [unbug/unbug.github.io#18] [Announcements]
# 一分钟读论文：《Montage：基于神经网络语言模型 (NNLM) 实现 JS 引擎模糊测试器》

基于神经网络语言模型 (NNLM) 实现 JS 引擎模糊测试器被证实有效。韩国科学技术院（KAIST）的一篇论文《Montage: A Neural Network Language Model-Guided JavaScript Engine Fuzzer》， 首次提出了基于神经网络语言模型 (NNLM) 实现 JS 引擎模糊测试器： Montage。 论文提出了一种将 JS 测试用例的层次结构和这些结构之间的关系建模为一系列片段的新算法。 将 AST 编码为片段序列使 Montage 能够使用 LSTM 模型学习片段之间的关系。 技术的关键方面是将 JS 抽象语法树 (AST) 转换为可以直接训练流行的 NNLM 的 AST 子树序列 试验用例结果表明 Montage 在最新的 JS 引擎中发现了37个真实 bug，有3个是公共漏洞和暴露（CVE），证明了它能有效发现 JS 引擎的 bug。

https://unbug.github.io/Montage-A-Neural-Network-Language-Model-Guided-JavaScript-Engine-Fuzzer/

**Comment by unbug** [comment]
基于 AI 实现的模糊测试器最大的优势就是不舍边界，它不受平台和语言的限制，真正实现跨平台测试，它将是全能的QA
